metau-capstone/StudyViewController.m

```diff
@@ -107,10 +98,23 @@ - (void)viewDidLoad {
                         PFQuery *query = [PFQuery queryWithClassName:@"Flashcard" predicate:predicate];
                         
                         // Fetch data for cards asynchronously
-                        [query findObjectsInBackgroundWithBlock:^(NSArray *cards, NSError *error) {
+                        [query findObjectsInBackgroundWithBlock:^(NSArray<Flashcard *> *cards, NSError *error) {
                             if (cards != nil) {
                                 self.arrayOfCards = cards;
-                                self.counter = 0;
+                                NSLog(@"phase 1");
+                                self.counter  = 0;
+                                if ([userObject[@"phaseNum"] isEqualToNumber:@(4)]) {
+                                    // Set toBeReviewed to be true for all card
+                                    for (Flashcard * cardToBeReviewed in self.arrayOfCards) {
+                                        cardToBeReviewed.toBeReviewed = YES;
+                                        [cardToBeReviewed saveInBackground];
+                                    }
+                                    userObject[@"phaseNum"] = @2;
+                                    [userObject saveInBackground];
+                                    
+                                    userObject[@"startedReview"] = @YES;
+                                    [userObject saveInBackground];
```

Comment:
I assume this `saveInBackground` method trigger a api call to update the database; if so, you want to avoid double calls in the same block, instead you should only trigger the call when all the states have been updated and trigger the call to update on the last step